### ISSUE TITLE
Usage of unitialized variable `p_keycode`

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#52488]
### Godot version

4.0.dev.custom_build. 24f562bd1

### System information

Not needed

### Issue description

Godot(compiled with `scons -j6 use_llvm=yes use_asan=yes`) shows usage of unitialized variable when opening editor

https://github.com/godotengine/godot/blob/d308a0a148bb18114c9ea81410d6b3ed08469e8a/editor/editor_settings.cpp#L1539

```
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior editor/editor_settings.cpp:1544:30 in 
editor/editor_settings.cpp:1545:28: runtime error: load of value 268435534, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior editor/editor_settings.cpp:1545:28 in 
editor/editor_settings.cpp:1546:29: runtime error: load of value 268435534, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior editor/editor_settings.cpp:1546:29 in 
editor/editor_settings.cpp:1547:29: runtime error: load of value 268435534, which is not a valid value for type 'Key'
```
https://github.com/godotengine/godot/blob/d308a0a148bb18114c9ea81410d6b3ed08469e8a/core/input/input_event.cpp#L393

```
core/input/input_event.cpp:393:18: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:393:18 in 
core/os/keyboard.h:345:20: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/os/keyboard.h:345:20 in 
core/input/input_event.cpp:394:18: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:394:18 in 
core/input/input_event.cpp:396:6: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:396:6 in 
core/input/input_event.cpp:399:6: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:399:6 in 
core/input/input_event.cpp:402:6: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:402:6 in 
core/input/input_event.cpp:405:6: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:405:6 in 
core/input/input_event.cpp:408:6: runtime error: load of value 50331650, which is not a valid value for type 'Key'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/input/input_event.cpp:408:6 in 
core/os/keyboard.h:341:20: runtime error: load of value 268435546, which is not a valid value for type 'Key'
```

### Steps to reproduce

Not needed

**Comment by kdiduk** [comment]
I can take care of this issue. Please, assign it to me.

**Comment by bruvzg** [comment]
This should be fixed by #51684, it's not uninitialized but incorrect cast from int (result of boolean op on enum members) to enum type.

**Comment by qarmin** [comment]
Yup, 
This is duplicate of #51641
